code/game/objects/objs.dm

```diff
@@ -138,19 +138,23 @@
 		else
 			obj_flags &= ~IN_USE
 
-/obj/proc/updateDialog()
+/obj/proc/updateDialog(update_viewers = TRUE,update_ais = TRUE)
 	// Check that people are actually using the machine. If not, don't update anymore.
 	if(obj_flags & IN_USE)
-		var/list/nearby = viewers(1, src)
-		var/is_in_use = 0
-		for(var/mob/M in nearby)
-			if ((M.client && M.machine == src))
-				is_in_use = 1
-				src.interact(M)
-		var/ai_in_use = AutoUpdateAI(src)
-
-		if(!ai_in_use && !is_in_use)
-			obj_flags &= ~IN_USE
+		var/is_in_use = FALSE
+		if(update_viewers)
+			var/list/nearby = viewers(1, src)
+			for(var/mob/M in nearby)

```

Comment:
IIRC, `viewers()` gets faster if you don't save it out to a list, as in `for(var/mob/M in viewers(1, src))`